File: consumer.py
import pika

# 1. Connect to RabbitMQ
connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='localhost')
)

# 2. Open a channel
channel = connection.channel()

# 3. Function that runs when message arrives
def callback(ch, method, properties, body):
    print("✅ Received:", body.decode())

# 4. Tell RabbitMQ to give messages from queue
channel.basic_consume(
    queue='demo-pc',
    on_message_callback=callback,
    auto_ack=True
)

print("⏳ Waiting for messages...")
channel.start_consuming()

File: Consumer.cs
using System;
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;

class Program
{
    static void Main(string[] args)
    {
        // Connect to RabbitMQ
        var factory = new ConnectionFactory()
        {
            HostName = "localhost"
        };

        var connection = factory.CreateConnection();
        var channel = connection.CreateModel();

        // Create consumer
        var consumer = new EventingBasicConsumer(channel);

        consumer.Received += (model, ea) =>
        {
            var body = ea.Body.ToArray();
            var message = Encoding.UTF8.GetString(body);

            Console.WriteLine("✅ Received: " + message);
        };

        // Consume messages from queue
        channel.BasicConsume(
            queue: "demo-pc",
            autoAck: true,
            consumer: consumer
        );

        Console.WriteLine("⏳ Waiting for messages...");
        Console.ReadLine();
    }
}
